app/Console/Commands/Indexer/IncrementalIndexer.php

```diff
@@ -0,0 +1,226 @@
+            $this->info("Resuming from index: $index");
+        }
+
+        if ($index > 0 && !isset($this->ids[$index]))
+        {
+            $index = 0;
+            $this->warn('Invalid index; set back to 0');
+        }
+
+        Storage::put("indexer/incremental/{$mediaType}_resume.save", 0);
+
+        $this->info("$idCount $mediaType entries available");
+        $ids = array_merge($idsToFetch['sfw'], $idsToFetch['nsfw']);
+
+        for ($i = $index; $i <= ($idCount - 1); $i++)
+        {
+            if ($this->cancelled)
+            {
+                return;
+            }
+
+            $id = $ids[$index];
+
+            $url = env('APP_URL') . "/v4/$mediaType/$id";
+            $this->info("Indexing/Updating " . ($i + 1) . "/$idCount $url [MAL ID: $id]");
+
+            try
+            {
+                $response = json_decode(file_get_contents($url), true);
+                if (!isset($response['error']) || $response['status'] == 404)
+                {
+                    continue;
+                }
+
+                $this->error("[SKIPPED] Failed to fetch $url - {$response['error']}");
+            }
+            catch (\Exception)
+            {
+                $this->warn("[SKIPPED] Failed to fetch $url");
+                $failedIds[] = $id;
+                Storage::put("indexer/incremental/$mediaType.failed", json_encode($failedIds));
+            }
+
+            $success[] = $id;
+            Storage::put("indexer/incremental/{$mediaType}_resume.save", $index);
+        }
+
+        Storage::delete("indexer/incremental/{$mediaType}_resume.save");
+
+        $this->info("--- Indexing of $mediaType is complete.");
+        $this->info(count($success) . ' entries indexed or updated.');
+        if (count($failedIds) > 0)
+        {
+            $this->info(count($failedIds) . ' entries failed to index or update. Re-run with --failed to requeue failed entries only.');
+        }
+
+        // finalize the latest state
+        Storage::move("indexer/incremental/$mediaType.json.tmp", "indexer/incremental/$mediaType.json");
+    }
+
+    public function handle(): int
+    {
+        // validate inputs
+        $validator = Validator::make(
+            [
+                'mediaType' => $this->argument('mediaType'),
+                'delay' => $this->option('delay'),
+                'resume' => $this->option('resume') ?? false,
+                'failed' => $this->option('failed') ?? false
+            ],
+            [
+                'mediaType' => 'required|in:anime,manga',
+                'delay' => 'integer|min:1',
+                'resume' => 'bool|prohibited_with:failed',
+                'failed' => 'bool|prohibited_with:resume'
+            ]
+        );
+
+        if ($validator->fails()) {
+            $this->error($validator->errors()->toJson());
+            return 1;
+        }
+
+        // we want to handle signals from the OS
+        $this->trap(SIGTERM, fn () => $this->cancelled = true);
```

Comment:
Would it be appropriate to use Symfony ConsoleEvents here? And subscribe to both SIGTERM and SIGINT?

https://jolicode.com/blog/handling-signal-with-symfony-command
https://symfony.com/doc/current/components/console/events.html

Reply:
@pushrbx You're right. That's odd since they're using symfony's console package for this 🤔 
Marking as resolved.